app/src/main/java/org/hackillinois/android/view/navigationdrawer/ScannerFragment.kt

```diff
@@ -0,0 +1,195 @@
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import com.google.zxing.ResultPoint
+import com.journeyapps.barcodescanner.BarcodeCallback
+import com.journeyapps.barcodescanner.BarcodeResult
+import kotlinx.android.synthetic.main.fragment_scanner.*
+import kotlinx.android.synthetic.main.fragment_scanner.view.*
+import org.hackillinois.android.model.CheckIn.CheckIn
+import org.hackillinois.android.model.Event.UserEventPair
+import org.hackillinois.android.viewmodel.ScannerViewModel
+
+
+class ScannerFragment : Fragment() {
+    var lastSuccessfullyScannedUser: String = ""
+    val PERMISSIONS_REQUEST_ACCESS_CAMERA = 0
+    lateinit var viewModel: ScannerViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(this).get(ScannerViewModel::class.java)
+        viewModel.init()
+        viewModel.eventsListLiveData.observe(this, Observer { updateEventList(it) })
+
+        // We will only observe the change in lastScanWasSuccessful, as if that changes,
+        // so will the scanned userId
+        viewModel.lastScanWasSuccessful.observe(this, Observer { processLastScan(it) })
+
+        viewModel.shouldDisplayOverrideSwitch.observe(this, Observer { updateOverrideSwitchVisibility(it) })
+    }
+
+    /**
+     * Make the switch invisible when not relevant to the event.
+     */
+    private fun updateOverrideSwitchVisibility(it: Boolean?) {
+        staffOverrideSwitch.visibility = when (it) {
+            true -> AdapterView.VISIBLE
+            false -> AdapterView.GONE
+            null -> AdapterView.GONE
+        }
+    }
+
+    /**
+     * Called when scanner scans something.
+     */
+    private var onQrCodeScan: BarcodeCallback = object : BarcodeCallback {
+        override fun possibleResultPoints(resultPoints: MutableList<ResultPoint>?) {
+        }
+
+        override fun barcodeResult(result: BarcodeResult) {
+            // Prevent duplicate scans
+            if (result.text == null || result.text == lastSuccessfullyScannedUser) {
+                return
+            }
+
+            if (lastSuccessfullyScannedUser == "") {
+                lastSuccessfullyScannedUser = result.text
+            }
+
+            Log.d("ScanEvent", "Scanned text: ${result.text}")
+            Toast.makeText(context, lastSuccessfullyScannedUser, Toast.LENGTH_LONG).show()
+
+            // Event that user is being scanned in for
+            var eventName: String = eventListView.selectedItem.toString()
+
+            // User's ID
+            var userId: String = getUserIdFromQrString(lastSuccessfullyScannedUser)
+
+
+            Log.d("ScanEvent", "User ID: ${userId}")
+            Log.d("ScanEvent", "Event: ${eventName}")
+
+            // Check-in is a special event in the HackIllinois API
+            if (eventName == "Check In") {
+                var override = staffOverrideSwitch.isChecked
```

Comment:
override is a keyword and probably shouldn't be used

Reply:
How about `staffOverride`?